rtos/source/TARGET_CORTEX/mbed_rtos_rtx.c

```diff
@@ -114,6 +114,9 @@ MBED_NORETURN void mbed_rtos_start()
     }
 
     osKernelStart();
+#if MBED_TRAP_ERRORS_ENABLED
```

Comment:
https://github.com/ARMmbed/mbed-os/pull/11470#issuecomment-530772790

Reply:
Previously `mbed_error` wasn't marked "noreturn", so I can imagine compilers complaining about returning? It might have complained with osRtxErrorNotify being "noreturn" itself.  Maybe the loops were to shut that up. 

Or perhaps people were just being over-cautious - not trusting it.

(With the "noreturn" marking, the compiler will actually not be making it possible for `mbed_error` to return anyway - it'll crash if it did)